lib/resty/upstream/healthcheck.lua

```diff
@@ -214,64 +211,53 @@ local function check_peer(ctx, id, peer, is_backup)
         return
     end
 
+    -- define shortcut error function
+    local report_error = function(...) 
```

Comment:
This is expensive in that we create a new Lua closure (a GC object) upon every `check_peer` function invocation, which is considered the "hot code path".